fenix/app/src/main/java/org/mozilla/fenix/settings/DebugAddTabsSettingsFragment.kt

```diff
@@ -0,0 +1,213 @@
+
+package org.mozilla.fenix.settings
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.fragment.app.Fragment
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Adds tabs for debug purposes
+ */
+class DebugAddTabsSettingsFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setContent {
+                FirefoxTheme {
+                    DebugAddTabsScreen()
+                }
+            }
+        }
+    }
+
+    @Composable
+    private fun DebugAddTabsScreen() {
+        val currentTabCount by components.core.store.observeAsState(
+            initialValue = 0,
+            map = { it.tabs.size },
+        )
+
+        Column(modifier = Modifier
+            .padding(8.dp),
+        ) {
+            CurrentTabCount(currentTabCount)
+            Spacer(modifier = Modifier.size(8.dp))
+            AddTabsButton(store = components.core.store)
+        }
+
+    }
+    
+    @Composable
+    private fun CurrentTabCount(count: Int) {
+        Row(
+            modifier = Modifier.padding(16.dp),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Text(
+                text = stringResource(R.string.debug_add_tabs_count),
+                color = FirefoxTheme.colors.textPrimary,
+                style = FirefoxTheme.typography.headline7,
+                modifier = Modifier.padding(4.dp)
+            )
+
+            Text(
+                text = count.toString(),
+                color = FirefoxTheme.colors.textPrimary,
+                modifier = Modifier.padding(4.dp)
+            )
+        }
+    }
+
+    @Composable
+    private fun AddTabsButton(store: BrowserStore) {
```

Comment:
Instead of passing down the store, could pass function that could be invoked with the list of urls
`onClick(urls: List<String>) -> Uni`

So the caller can convert that to tabs and dispatch the action, we won't also repeat the private param in that case.